src/Zicht/Itertools/lib/SliceIterator.php

```diff
@@ -83,6 +83,9 @@ public function __construct(\Iterator $iterable, $start, $end = null)
     {
         if ($start < 0 || $end < 0) {
             $length = iterator_count($iterable);
+        } else {
+            // length is not needed.  still, we will define it for code cleanliness
+            $length = 0;
```

Comment:
You could define length at the start of your function. or simply return iterator count and return 0 at the end.